Build process ignores the $CC and $CXX environment variables

I'm pretty sure this is an issue that spans all of pret's makefiles and build scripts, but here's where I found it, so here's where I'm opening it.

The values of the `$CC` and `$CXX` environment variables is completely ignored; the names of `gcc` and `g++` are hardcoded. In systems like mine, with several versions of those tools, this causes the wrong tool to be invoked, leading to build failures.

On testing, this claim does not hold water.  (Ignore the error, I don't have libpng for the mingw32 target in this environment.)

```
pokeemerald$ make tools CC=i686-w64-mingw32-gcc CXX=i686-w64-mingw32-g++
i686-w64-mingw32-g++ -Wall -std=c++11 -O2 json11.cpp mapjson.cpp -o mapjson
i686-w64-mingw32-gcc -Wall -Wextra -Wno-switch -Werror -std=c11 -O2 main.c extended.c -o aif2pcm  -lm
i686-w64-mingw32-g++ -Wall -std=c++11 -O2 -I . jsonproc.cpp -o jsonproc
i686-w64-mingw32-g++ -std=c++11 -O2 -Wall -Wno-switch -Werror main.cpp sym_file.cpp elf.cpp -o ramscrgen
i686-w64-mingw32-g++ -Wall -Werror -std=c++11 -O2 scaninc.cpp c_file.cpp asm_file.cpp source_file.cpp -o scaninc
i686-w64-mingw32-gcc -Wall -Wextra -Werror -std=c11 -O2 bin2c.c -o bin2c
i686-w64-mingw32-gcc -Wall -Wextra -Werror -std=c11 -O2 -DPNG_SKIP_SETJMP_CHECK main.c convert_png.c util.c font.c -o rsfont  -lpng -lz
convert_png.c:5:10: fatal error: png.h: No such file or directory
 #include <png.h>
          ^~~~~~~
compilation terminated.
Makefile:12: recipe for target 'rsfont' failed
make[1]: *** [rsfont] Error 1
Makefile:138: recipe for target 'tools/rsfont' failed
make: *** [tools/rsfont] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,17 @@
 TOOLCHAIN := $(DEVKITARM)
+
+ifeq ($(CC),)
+HOSTCC := gcc
+else
+HOSTCC := $(CC)
+endif
+
+ifeq ($(CXX),)
+HOSTCXX := g++
+else
+HOSTCXX := $(CXX)
+endif
+
 ifneq (,$(wildcard $(TOOLCHAIN)/base_tools))
 include $(TOOLCHAIN)/base_tools
 else
@@ -136,7 +149,7 @@ all: rom
 tools: $(TOOLDIRS)
 
 $(TOOLDIRS):
-	@$(MAKE) -C $@
+	@$(MAKE) -C $@ CC=$(HOSTCC) CXX=$(HOSTCXX)
 
 rom: $(ROM)
 

--- a/tools/aif2pcm/Makefile
+++ b/tools/aif2pcm/Makefile
@@ -1,4 +1,4 @@
-CC = gcc
+CC ?= gcc
 
 CFLAGS = -Wall -Wextra -Wno-switch -Werror -std=c11 -O2
 

--- a/tools/bin2c/Makefile
+++ b/tools/bin2c/Makefile
@@ -1,4 +1,4 @@
-CC = gcc
+CC ?= gcc
 
 CFLAGS = -Wall -Wextra -Werror -std=c11 -O2
 

--- a/tools/gbafix/Makefile
+++ b/tools/gbafix/Makefile
@@ -1,4 +1,4 @@
-CC = gcc
+CC ?= gcc
 .PHONY: all clean
 
 SRCS = gbafix.c

--- a/tools/gbagfx/Makefile
+++ b/tools/gbagfx/Makefile
@@ -1,4 +1,4 @@
-CC = gcc
+CC ?= gcc
 
 CFLAGS = -Wall -Wextra -Werror -std=c11 -O2 -DPNG_SKIP_SETJMP_CHECK
 

--- a/tools/jsonproc/Makefile
+++ b/tools/jsonproc/Makefile
@@ -1,4 +1,4 @@
-CXX := g++
+CXX ?= g++
 
 CXXFLAGS := -Wall -std=c++11 -O2
 

--- a/tools/mapjson/Makefile
+++ b/tools/mapjson/Makefile
@@ -1,4 +1,4 @@
-CXX := g++
+CXX ?= g++
 
 CXXFLAGS := -Wall -std=c++11 -O2
 

--- a/tools/mid2agb/Makefile
+++ b/tools/mid2agb/Makefile
@@ -1,4 +1,4 @@
-CXX := g++
+CXX ?= g++
 
 CXXFLAGS := -std=c++11 -O2 -Wall -Wno-switch -Werror
 

--- a/tools/preproc/Makefile
+++ b/tools/preproc/Makefile
@@ -1,4 +1,4 @@
-CXX := g++
+CXX ?= g++
 
 CXXFLAGS := -std=c++11 -O2 -Wall -Wno-switch -Werror
 

--- a/tools/ramscrgen/Makefile
+++ b/tools/ramscrgen/Makefile
@@ -1,4 +1,4 @@
-CXX := g++
+CXX ?= g++
 
 CXXFLAGS := -std=c++11 -O2 -Wall -Wno-switch -Werror
 

--- a/tools/rsfont/Makefile
+++ b/tools/rsfont/Makefile
@@ -1,4 +1,4 @@
-CC = gcc
+CC ?= gcc
 
 CFLAGS = -Wall -Wextra -Werror -std=c11 -O2 -DPNG_SKIP_SETJMP_CHECK
 

--- a/tools/scaninc/Makefile
+++ b/tools/scaninc/Makefile
@@ -1,4 +1,4 @@
-CXX = g++
+CXX ?= g++
 
 CXXFLAGS = -Wall -Werror -std=c++11 -O2
 

